outlook/src/taskpane/components/Project/TasksSection/TasksSection.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import Partner from "../../../../classes/Partner";
+import Project from "../../../../classes/Project";
+import Task from "../../../../classes/Task";
+import {ContentType, HttpVerb, sendHttpRequest} from "../../../../utils/httpRequest";
+import {_t} from "../../../../utils/Translator";
+import CollapseSection from "../../CollapseSection/CollapseSection";
+import TaskListItem from "../TaskList/TaskListItem";
+import api from "../../../api";
+import AppContext from '../../AppContext';
+import {Callout, DirectionalHint} from "office-ui-fabric-react";
+import SelectProjectDropdown from "./SelectProjectDropdown";
+
+type TasksSectionProps = {
+    partner: Partner;
+}
+
+type TasksSectionState = {
+    tasks: Task[];
+    isCollapsed: boolean;
+    isProjectCalloutOpen: boolean;
+    key: number;
+}
+
+class TasksSection extends React.Component<TasksSectionProps, TasksSectionState> {
+
+    constructor(props, context) {
+        super(props, context);
+        let isCollapsed = true;
+        if (props.partner.tasks && props.partner.tasks.length > 0) {
+            isCollapsed = false;
+        }
+        this.state = {
+            tasks: this.props.partner.tasks, isCollapsed: isCollapsed,
+            isProjectCalloutOpen: false, key: Math.random()
+        };
+    }
+
+    private toggleProjectCallout = () => {
+        this.setState({isProjectCalloutOpen: !this.state.isProjectCalloutOpen});
+    }
+
+    private closeProjectCallout = () => {
+        this.setState({isProjectCalloutOpen: false});
+    }
+
+    private onCollapseButtonClick = () => {
+        this.setState({isCollapsed: !this.state.isCollapsed});
+    };
+
+    private onProjectSelected = (project: Project): void => {
+        this.closeProjectCallout();
+        this.createTask(project);
+    }
+
+    private createTask = (project: Project) => {
+        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, (result) => {
+            const message = result.value.split('<div id="x_appendonsend"></div>')[0]; // Remove the history and only log the most recent message.
+            const subject = Office.context.mailbox.item.subject;
+
+            const requestJson = {
+                partner_id: this.props.partner.id,
+                email_body: message,
+                email_subject: subject,
+                project_id: project.id,
+            }
+
+            const createTaskRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.createTask,
+                ContentType.Json,
+                this.context.getConnectionToken(), requestJson, true);
+            createTaskRequest.promise.then(response => {
+                const parsed = JSON.parse(response);
+                if (parsed['error']) {
+                    this.context.showTopBarMessage();
+                    return;
+                } else {
+                    const cids = this.context.getUserCompaniesString();
```

Comment:
Are we sure it is aligned with chosen project for the task? Normally yes actually as you display projects based on current companies (I suppose) ?

Reply:
It has been decided to tick all existing companies by default, so I think that we can be assured that the user won't get an error when redirected to the project.